sphinxcontrib/openapi/_lib2to3.py

```diff
@@ -0,0 +1,366 @@
+
+            if parameters:
+                oas3_operation["parameters"] = parameters
+
+            # According to OAS 2, 'body' and 'formData' are mutually exclusive
+            # parameters. Therefore, we can assign either one in one statement.
+            if request_body or request_form:
+                oas3_operation["requestBody"] = request_body or request_form
+
+        oas3_operation["responses"] = self.convert_responses(
+            oas2_operation["responses"], oas2_produces=oas2_operation.get("produces")
+        )
+        return oas3_operation
+
+    def convert_request_body(self, oas2_operation):
+        oas3_request_body = {}
+
+        for parameter in oas2_operation["parameters"]:
+            if parameter["in"] == "body":
+                oas3_request_body = _get_properties(
+                    parameter, {"description", "required"}
+                )
+                mimetypes = oas2_operation.get("consumes") or ["*/*"]
+                content = {}
+                if "schema" in parameter:
+                    content["schema"] = parameter["schema"]
+                oas3_request_body["content"] = {mimetypes[0]: content}
+                break
+
+        return oas3_request_body
+
+    def convert_request_body_formdata(self, oas2_operation):
+        mimetypes = {
+            "application/x-www-form-urlencoded",
+            "multipart/form-data",
+        }.intersection(
+            oas2_operation.get("consumes", ["application/x-www-form-urlencoded"])
+        )
+        mediatype = {}
+
+        for parameter in oas2_operation["parameters"]:
+            if parameter["in"] == "formData":
+                schema = mediatype.setdefault(
+                    "schema", {"type": "object", "properties": {}}
+                )
+                schema["properties"][parameter["name"]] = _get_schema_properties(
+                    parameter, except_for={"name", "in", "required"}
+                )
+
+                if parameter.get("required"):
+                    schema.setdefault("required", []).append(parameter["name"])
+
+                # Excerpt from OpenAPI 2.x spec:
+                #
+                # > If type is "file", the consumes MUST be either
+                # > "multipart/form-data", "application/x-www-form-urlencoded"
+                # > or both and the parameter MUST be in "formData".
+                #
+                # This is weird since HTTP does not allow file uploading in
+                # 'application/x-www-form-urlencoded'. Moreover, Swagger
+                # editor complains if 'file' is detected and there's no
+                # 'multipart/form-data' in `consumes'.
+                if parameter["type"] == "file":
+                    mimetypes = ["multipart/form-data"]
+
+        if not mediatype:
+            return None
+
+        return {"content": {mimetype: mediatype for mimetype in mimetypes}}
+
+    def convert_parameters(self, oas2_parameters):
```

Comment:
It looks like parameters are allowed to have vendor extensions as well: https://swagger.io/docs/specification/openapi-extensions/

Or am I missing something and this is covered one layer above?

Reply:
`Parameter Object` supports vendor extensions, `Paramters Object` (plural) does not. `Parameters Object` is a list and hence you cannot put vendor extensions in it.